described basic working of llm flows classifier (#12529)
RasaHQ/rasa: docs/docs/llms/llm-flow-classifier.mdx

<RasaDiscoveryBanner/>

> # LLMFlowClassifier

Reviewer: we do need a better name, but not today. i find this a bit confusing

* Slot setting, described by "SetSlot(slot_name, slot_value)". An example would be "SetSlot(recipient, Freddy)"
* Starting another flow, described by "StartFlow(flow_name)". An example would be "StartFlow(transfer_money)"
* Cancelling the current flow, describe by "CancelFlow()"

Write out the actions you want to take for the last user message, one per line.
Do not prematurely fill slots with abstract values.
Only use information provided by the user.
Strictly adhere to the provided action types above for starting flows and setting slots.
Focus on the last message and take it one step at a time.
Use the previous conversation steps only to aid understanding.
The action list:
```

The LLM might reply to this prompt with:
```
SetSlot(transfer_money_amount_of_money, 40$)
```
Which we would then turn into an `inform` intent, since no other flow needs to be
started and the information belongs to the question that was just asked. Also,
the `transfer_money_amount_of_money` entity would be present.

If the user had said "Sorry, I meant to John" instead, the LLM might have replied with
```
SetSlot(transfer_money_recipient, John)
```

> Which we then would turn into an `correction` intent, since it tries to set a slot that

Reviewer 2: ```suggestion
Which would then be turned into a `correction` intent, since it tries to set a slot that
```

> was asked earlier in the flow. Also, the `transfer_money_recipient` entity would be
> present.

Reviewer 2: ```suggestion
was asked earlier in the flow. The `transfer_money_recipient` entity would also be
detected.
```

If the user had said "How much do I still have in my account" instead, the LLM might
have replied with
```
StartFlow(check_balance)
```
Which we then would turn into a `check_balance` intent.

It is also possible to return multiple actions at the same time for the LLM, if the
user for example says "I want to send 40$ to John", the LLM should reply:
```
StartFlow(transfer_money)
SetSlot(transfer_money_recipient, John)
SetSlot(transfer_money_amount_of_money, 40$)
```

which would then be turned into the `transfer_money` intent as well as two entities.

Certain asks by the user and responses by the LLM are currently deemed as too complex

> and classified as such based on the composition of the action list.

Reviewer 2: can we explain a bit more about why we needed to do this? and can you explain what happens when something is classified as "too complex": is the result that the output of the classifier is ignored?

Author: Sure thing, users can technically come up with arbitrarily complex commands like "start 1000000 money transfers" but on a more benign site they might just try to do a lot of things at once "50$ to James. And then please tell me how much money I still have left, and my recent transactions". For now, we just say, let's take it step by step. Doesn't mean we can't allow for more complexity later. We might actually be able it, but I think it's also good if there's some limit and we communicate that instead of attempting and failing.

> 
> Certain asks by the user and responses by the LLM are currently deemed as too complex
> and classified as such based on the composition of the action list.
> 
> 
> 

Reviewer 2: ```suggestion

### Too complex requests

Some requests from the user and responses by the LLM are currently deemed as too complex
and classified as such based on the composition of the action list.
```
